src/renderer/services/eth2/client/prysm/prysm.ts

```diff
@@ -23,35 +22,27 @@ export class PrysmEth2ApiClient extends AbstractApiClient implements IValidatorB
 
     public constructor(options: IBeaconClientOptions) {
         super();
-        options.urlPrefix = `${options.urlPrefix}${apiPrefix}`;
-        this.url = options.urlPrefix;
+        options.baseUrl = `${options.baseUrl}${apiPrefix}`;
+        this.url = options.baseUrl;
         this.config = options.config;
         this.beacon = new PrysmBeaconApiClient(options);
         this.validator = new PrysmValidatorApiClient(options);
     }
 
-    public static getPrysmBeaconClient(url: string, network: string): PrysmEth2ApiClient | null {
-        const networkConfig = getNetworkConfig(network);
-        if (!networkConfig) {
-            return null;
-        }
-
-        return new PrysmEth2ApiClient({
-            urlPrefix: url,
-            config: networkConfig.eth2Config
-        });
-    }
-
     public async getVersion(): Promise<string> {
         return (await this.beacon.getClientVersion()).toString("ascii");
     }
 
-    public async isSyncing(): Promise<boolean> {
-        const result = await this.beacon.getSyncingStatus();
-        return (typeof result === "boolean") ? result : false;
-    }
-
-    public async getChainHeight(): Promise<string> {
-        return (await this.beacon.getChainHead()).headSlot;
+    public onNewChainHead(callbacks: Array<any>, params: Array<any>): void {
+        // const eventSource = new EventSource(`${this.url}${PrysmBeaconRoutes.CHAINHEAD_STREAM}`);
+        // eventSource.onmessage = (e: MessageEvent): void => {
+        //     console.log("Got message!", e);
+        //     callbacks.map(callback => callback(e.data as PrysmChainHeadStreamResponse));
+        // };
```

Comment:
Did they confirm this doesn't work?

Reply:
No, I didn't get any answer. However, I couldn't get it to work. It's possible that that is not working as I got some error regarding wrong MIME types and closed connections. I've only tried testing it through React using EventSource.

However, since you mentioned this is not available in Lighthouse, I figured no point in losing more time for the same thing.

Reply:
Sure, because lighthouse will add those apis afaik